From: Baoquan He <bhe@redhat.com>
Subject: arch/loongarch: clean up the left code and Kconfig item related to CRASH_CORE
Date: Fri, 29 Mar 2024 21:28:24 +0800

Patch series "crash: clean up outdated codes and doc related to
CRASH_CORE".

As Geert pointed out that there are still left codes and doc related to
CRASH_CORE in arch/loongarch and Documentation.  So make these two patches
to clean them up.  Thanks to Geert.


This patch (of 2):

In commit 85fcde402db1 ("kexec: split crashkernel reservation code out
from crash_core.c"), crashkernel reservation code is split out from
crash_core.c, and add CRASH_RESERVE to control it.  And also rename each
ARCH's <asm/crash_core.h> to <asm/crash_reserve.h> accordingly.

But the relevant part in loongarch is missed. Do it now.

Link: https://lkml.kernel.org/r/20240329132825.1102459-1-bhe@redhat.com
Link: https://lkml.kernel.org/r/20240329132825.1102459-2-bhe@redhat.com
Signed-off-by: Baoquan He <bhe@redhat.com>
Cc: Huacai Chen <chenhuacai@kernel.org>
Cc: WANG Xuerui <kernel@xen0n.name>
Cc: Geert Uytterhoeven <geert@linux-m68k.org>
Cc: Jonathan Corbet <corbet@lwn.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/loongarch/Kconfig                     |    2 +-
 arch/loongarch/include/asm/crash_core.h    |   12 ------------
 arch/loongarch/include/asm/crash_reserve.h |   12 ++++++++++++
 3 files changed, 13 insertions(+), 13 deletions(-)

--- a/arch/loongarch/include/asm/crash_core.h
+++ /dev/null
@@ -1,12 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0-only */
-#ifndef _LOONGARCH_CRASH_CORE_H
-#define _LOONGARCH_CRASH_CORE_H
-
-#define CRASH_ALIGN			SZ_2M
-
-#define CRASH_ADDR_LOW_MAX		SZ_4G
-#define CRASH_ADDR_HIGH_MAX		memblock_end_of_DRAM()
-
-extern phys_addr_t memblock_end_of_DRAM(void);
-
-#endif
--- /dev/null
+++ a/arch/loongarch/include/asm/crash_reserve.h
@@ -0,0 +1,12 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#ifndef _LOONGARCH_CRASH_RESERVE_H
+#define _LOONGARCH_CRASH_RESERVE_H
+
+#define CRASH_ALIGN			SZ_2M
+
+#define CRASH_ADDR_LOW_MAX		SZ_4G
+#define CRASH_ADDR_HIGH_MAX		memblock_end_of_DRAM()
+
+extern phys_addr_t memblock_end_of_DRAM(void);
+
+#endif
--- a/arch/loongarch/Kconfig~arch-loongarch-clean-up-the-left-code-and-kconfig-item-related-to-crash_core
+++ a/arch/loongarch/Kconfig
@@ -596,7 +596,7 @@ config ARCH_SELECTS_CRASH_DUMP
 	select RELOCATABLE
 
 config ARCH_HAS_GENERIC_CRASHKERNEL_RESERVATION
-	def_bool CRASH_CORE
+	def_bool CRASH_RESERVE
 
 config RELOCATABLE
 	bool "Relocatable kernel"
_
